### PR TITLE
Add docs for Circleci rerun failed tests feature

### PR DESCRIPTION
**Pull request by ArturT**
* Add docs for Circleci rerun failed tests feature
* Add info about advanced CircleCI features in the installation guide for the Ruby projects

# story

https://trello.com/c/XFXyRYPu

### CODE REVIEW

File: docusaurus/docs/ruby/circleci.md [hunk 1]
```diff
@@ -62,3 +62,46 @@ end
...
+      mkdir -p $CIRCLE_TEST_REPORTS
+
+      # split slow spec files by test examples
+      # https://docs.knapsackpro.com/ruby/split-by-test-examples/
+      export KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES=true
+
+      # highlight-start
+      # Use circleci CLI to find out if we need to run all tests or rerun failed tests
+      # We are telling circleci to split the tests across 1 node to get the full list of all tests for consideration. We leave the splitting to Knapsack Pro.
+      circleci tests glob "spec/**/*_spec.rb" | circleci tests run --index 0 --total 1 --command ">/tmp/tests_to_run.txt xargs echo" --verbose > /tmp/tests_to_run.txt
```
Comment (3v0k4) [inline docusaurus/docs/ruby/circleci.md:86]: I think that with `-n1` you won't need the `sed` later:

```suggestion
      circleci tests glob "spec/**/*_spec.rb" | circleci tests run --index 0 --total 1 --command ">/tmp/tests_to_run.txt xargs -n1 echo" --verbose > /tmp/tests_to_run.txt
```

File: docusaurus/docs/ruby/circleci.md [hunk 2]
```diff
@@ -62,3 +62,46 @@ end
...
+      export CIRCLE_TEST_REPORTS=/tmp/test-results
+      mkdir -p $CIRCLE_TEST_REPORTS
+
+      # split slow spec files by test examples
+      # https://docs.knapsackpro.com/ruby/split-by-test-examples/
+      export KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES=true
+
+      # highlight-start
+      # Use circleci CLI to find out if we need to run all tests or rerun failed tests
+      # We are telling circleci to split the tests across 1 node to get the full list of all tests for consideration. We leave the splitting to Knapsack Pro.
```
Comment (3v0k4) [inline docusaurus/docs/ruby/circleci.md:85]:
```suggestion
      # Retrieve the tests to run (either all tests or just the failed ones), and let Knapsack Pro split them optimally.
```

**Review by 3v0k4 — APPROVED** [review]
Fantastic that integrates that easily. Great job 🙏

This is a great first step. But what if we actually did it inside the gem, so that users don't have to set up anything?

File: docusaurus/docs/ruby/circleci.md [hunk 3]
```diff
@@ -62,3 +62,46 @@ end
 </Tabs>
 
 You can find a complete CircleCI configuration in [RSpec testing parallel jobs with CircleCI and JUnit XML report](https://docs.knapsackpro.com/2021/rspec-testing-parallel-jobs-with-circleci-and-junit-xml-report).
+
+## Rerun failed tests
+
+Use the [CircleCI rerun failed tests](https://circleci.com/docs/rerun-failed-tests/) feature with Knapsack Pro to only rerun a subset of tests instead of rerunning the entire test suite when a transient test failure arises.
+
+Thanks to that, you save time and CircleCI resources. See a [video demo](https://www.youtube.com/watch?v=gf2BK_ZukUg).
```
[the remark this answers is not in the document]
Comment (ArturT) [reply docusaurus/docs/ruby/circleci.md]: Good point. I cropped the video and increased the volume.

File: docusaurus/docs/ruby/circleci.md (hunk 1 above)
Comment (ArturT) [reply docusaurus/docs/ruby/circleci.md:88]: It works. Cool. 🚀

**Pull request merged**